app/src/sharedTest/java/org/oppia/android/app/parser/CustomBulletSpanTest.kt

```diff
@@ -0,0 +1,186 @@
+    val textView =
+      activityTestRule.activity.findViewById(R.id.test_html_content_text_view) as TextView
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      /* entityType= */ "",
+      /* entityId= */ "",
+      /* imageCenterAlign= */ true
+    )
+    val htmlResult: Spannable = htmlParser.parseOppiaHtml(
+      "<p>You should know the following before going on:<br></p>" +
+        "<ul><li>The counting numbers (1, 2, 3, 4, 5 ….)<br></li>" +
+        "<li>How to tell whether one counting number is bigger or " +
+        "smaller than another<br></li></ul>",
+      textView
+    )
+
+    /* Reference: https://medium.com/androiddevelopers/spantastic-text-styling-with-spans-17b0c16b4568#e345 */
+    val bulletSpans =
+      htmlResult.getSpans<CustomBulletSpan>(0, htmlResult.length, CustomBulletSpan::class.java)
+    Truth.assertThat(bulletSpans.size.toLong()).isEqualTo(2)
+
+    val bulletSpan0 = bulletSpans[0] as CustomBulletSpan
+    Truth.assertThat(bulletSpan0).isNotNull()
+
+    val bulletRadius = launchedActivity.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.bullet_radius
+    )
+    val spacingBeforeBullet = launchedActivity.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.spacing_before_bullet
+    )
+    val spacingBeforeText = launchedActivity.resources.getDimensionPixelSize(
+      org.oppia.android.util.R.dimen.spacing_before_text
+    )
+    val expectedMargin = spacingBeforeBullet + spacingBeforeText + 2 * bulletRadius
+
+    val bulletSpan0Margin = bulletSpan0.getLeadingMargin(true)
+    Truth.assertThat(bulletSpan0Margin).isEqualTo(expectedMargin)
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<TestApplication>().inject(this)
+  }
+
+  @Singleton
+  @Component(
+    modules = [
+      TestDispatcherModule::class, ApplicationModule::class,
+      LoggerModule::class, ContinueModule::class, FractionInputModule::class,
+      ItemSelectionInputModule::class, MultipleChoiceInputModule::class,
+      NumberWithUnitsRuleModule::class, NumericInputRuleModule::class, TextInputRuleModule::class,
+      DragDropSortInputModule::class, ImageClickInputModule::class, InteractionsModule::class,
+      GcsResourceModule::class, GlideImageLoaderModule::class, ImageParsingModule::class,
+      HtmlParserEntityTypeModule::class, QuestionModule::class, TestLogReportingModule::class,
+      TestAccessibilityModule::class, LogStorageModule::class, CachingTestModule::class,
+      PrimeTopicAssetsControllerModule::class, ExpirationMetaDataRetrieverModule::class,
+      ViewBindingShimModule::class, RatioInputModule::class,
+      ApplicationStartupListenerModule::class, LogUploadWorkerModule::class,
+      WorkManagerConfigurationModule::class, HintsAndSolutionConfigModule::class,
+      FirebaseLogUploaderModule::class
+    ]
+  )
+
+  interface TestApplicationComponent : ApplicationComponent {
```

Comment:
I don't think you need a custom test application or any of the dagger stuff here since you don't need to inject anything (after removing the HtmlParser dependency).

Reply:
@FareesHussain Please reply to this.

Reply:
Done

Reply:
@FareesHussain This as not been done as per Ben's suggestions. You can remove this entire TestApplicationComponent your tests should work.

Reply:
@rt4914 I've discussed this with ben
actually in CustomBulletSpanTest it requires context so we can't remove it

Reply:
What he meant was when i was using htmlParser within `CustomBulletSpanTest` he said to remove it and create test cases independent of htmlParser and asked to remove this thinking that the `CustomBulletSpan` does'nt require any dagger injections
But the `CustomBulletSpan` requires context so this can't be removed

Reply:
@FareesHussain Actually I have tried it locally and all tests pass (not sure about CI).

Try this
`private var context: Context = ApplicationProvider.getApplicationContext()`

Reply:
@rt4914 I've tried this but this doesn't work with `Robolectric`

Reply:
This only works with Espresso

Reply:
@rt4914 I've created a gist for better reference of what i've tried 
https://gist.github.com/FareesHussain/3f54c7cc39f887278dd345c408c377ed

in this only one test case i passed and all other tests give a error
![image](https://user-images.githubusercontent.com/54740946/103400497-4835c780-4b6b-11eb-9991-91f4743dbd52.png)

Reply:
@FareesHussain Added on comment on your gist. Technically it should work and it is working for me.
<img width="1603" alt="Screenshot 2021-01-04 at 12 20 39 PM" src="https://user-images.githubusercontent.com/9396084/103508583-43348900-4e87-11eb-9191-4811b7bd56b2.png">

Reply:
Done

Reply:
Hmm, keeping the test application is fine (it should be avoidable here even for Espresso runs).